### PR TITLE
Timestamp generator

### PR DESCRIPTION
**Pull request by jorgebay**

### CODE REVIEW

File: src/Cassandra.IntegrationTests/Core/TimestampTests.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                CREATE TABLE {0}(
+                id uuid PRIMARY KEY,
+                label text);", "tbl_timestamp_test"));
+                var now = DateTimeOffset.Parse("2017-01-01T00:00:00.000+00:00");
+                var id = Guid.NewGuid();
+                var stmt = new SimpleStatement("INSERT INTO tbl_timestamp_test (id, label) VALUES (?,?) USING TIMESTAMP ?",
+                    id, "test label", now.Ticks);
+                session.Execute(stmt);
+                // The driver should use the generator against C* 2.1+
+                Assert.AreEqual(CassandraVersion < new Version(2, 1) ? 0 : 4, generator.GetCounter());
```
Comment (landim) [inline src/Cassandra.IntegrationTests/Core/TimestampTests.cs]: In that case, which "USING TIMESTAMP" at cql, the timestamp generator must tick or just ignore?

Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Core/TimestampTests.cs]: If the protocol version is lower than 3, the driver will not generate a new timestamp.
The driver doesn't parse the query and it's unaware if the query contains `USING TIMESTAMP` (provided by the user only).

File: src/Cassandra.IntegrationTests/Core/TimestampTests.cs [hunk 2]
```diff
@@ -33,6 +34,59 @@ public void Cluster_Uses_Provided_Timestamp_Generator()
             }
         }
 
+        [Test]
+        public void Should_Use_Statement_Timestamp_Precedence_Over_Cluster_Timestamp_Generator()
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Core/TimestampTests.cs]: Nice!

File: src/Cassandra.IntegrationTests/Core/TimestampTests.cs (hunk 1 above)
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Core/TimestampTests.cs]: We could retrieve the stored timestamp using `writetime()` CQL function to know that `USING TIMESTAMP ?` is the value that was used but that would be a test for the server behaviour (not the driver).

I say we should remove this test.

**Pull request merged**